src/pstore/pstore.c

```diff
@@ -171,153 +171,146 @@ static int move_file(PStoreEntry *pe, const char *subdir) {
                         continue;
-                }
-
-                /* See if this file is one of a related group of files
-                 * in order to reconstruct dmesg */
-
-                /* When dmesg is written into pstore, it is done so in
-                 * small chunks, whatever the exchange buffer size is
-                 * with the underlying pstore backend (ie. EFI may be
-                 * ~2KiB), which means an example pstore with approximately
-                 * 64KB of storage may have up to roughly 32 dmesg files
-                 * that could be related, depending upon the size of the
-                 * original dmesg.
-                 *
-                 * Here we look at the dmesg filename and try to discern
-                 * if files are part of a related group, meaning the same
-                 * original dmesg.
-                 *
-                 * The two known pstore backends are EFI and ERST. These
-                 * backends store data in the Common Platform Error
-                 * Record, CPER, format. The dmesg- filename contains the
-                 * CPER record id, a 64bit number (in decimal notation).
-                 * In Linux, the record id is encoded with two digits for
-                 * the dmesg part (chunk) number and 3 digits for the
-                 * count number. So allowing an additional digit to
-                 * compensate for advancing time, this code ignores the
-                 * last six digits of the filename in determining the
-                 * record id.
-                 *
-                 * For the EFI backend, the record id encodes an id in the
-                 * upper 32 bits, and a timestamp in the lower 32-bits.
-                 * So ignoring the least significant 6 digits has proven
-                 * to generally identify related dmesg entries.  */
-#define PSTORE_FILENAME_IGNORE 6
-
-                /* determine common portion of record id */
-                ++p; /* move beyond dmesg- */
+                ++p; /* move beyond dmesg-xxx- */
                 plen = strlen(p);
-                if (plen > PSTORE_FILENAME_IGNORE) {
-                        pe_id = memdup_suffix0(p, plen - PSTORE_FILENAME_IGNORE);
-                        if (!pe_id) {
+                if (plen < 6)
+                        continue;
+
+                if (startswith(pe->dirent.d_name, "dmesg-efi-")) {
+                        /* For the EFI backend, the 3 least significant digits
+                         * of record id encodes a "count" number, the next 2
+                         * least significant digits for the dmesg part (chunk)
+                         * number, and the remaining digits as the timestamp.
+                         * See linux/drivers/firmware/efi/efi-pstore.c in
+                         * efi_pstore_write(). */
+
+                        /* Extract base record id */
+                        subdir1 = memdup_suffix0(p, plen-5);
```

Comment:
Why `memdup_suffix0` is used instead of `strndup()`?

Reply:
I dont recall why, but I have changed to strndup()